api/src/routes/edit.ts

```diff
@@ -82,6 +102,37 @@ async function getChildren(req, res) {
     res.json(response);
 }
 
+function uploadFile(req, res, next) {
+    const { pid } = req.params;
+    const form = formidable({ multiples: true });
+
+    form.parse(req, async (err, fields, files) => {
+        if (err) {
+            next(err);
+            return;
+        }
+        try {
+            const datastream = Datastream.getInstance();
+            const { stream } = fields;
+            const { filepath, mimetype } = files?.file;
+            await datastream.uploadFile(pid, stream, filepath, mimetype);
+            res.status(200).send("Upload success");
+        } catch (error) {
+            res.status(500).send(error.message);
+        }
+    });
+}
+
+edit.get("/object/details/:pid", requireToken, pidSanitizer, async function (req, res) {
```

Comment:
This is a duplicate route name -- see line 87/138 below.

Reply:
Made change to /modelsdatastreams/

Reply:
Thanks! In the future, we may want to look into whether we should combine some of these things to reduce the number of API calls, but for now this makes sense. Of course I realize these are the sorts of issues that GraphQL can address, though best to keep it simple for now.